rfcs/0002-typography-scaling.md

```diff
@@ -0,0 +1,162 @@
+
+>📝 Note: Here we're rounding-off all the numbers for the sake of simplicity and to align the content according to the grid-system.
+
+# Drawbacks/Constraints
+Few things to consider here:
+- As we are moving towards adaptive-layout so we need to consider - should we also support responsive layout? For both cases, I don't see any challenges as the typography system remains the same.
+- How will this be managed on the tech-side? This might create more work as we'll have to maintain four scales,
+  - Landing/Marketing page (mobile & desktop/tablets)
+  - Product/Dashboard (mobile & desktop/tablets)
+
+
+# Alternatives
+Few other things we considered:
+
+### **Multiple type-scaling across breakpoints**
+In this type of scaling the modular scale changes on every breakpoint. For example: **mobiles** will use `Minor Third (1.2)` and **desktops/tablets** will use `Major Second (1.125)`.
+
+This doesn't suit our product use case, as there will be a big difference between all the type levels. For example:
+
+![Scaling: Multiple type scales ratio](https://miro.medium.com/max/4800/1*Gd6HOQUgDSisN0Ebubgk5g.png)
+
+### **No type-scaling across breakpoints**
+As the name suggests, there will be no scaling across all breakpoints, which is not at all recommended as all the font sizes will remain the same across all breakpoints, which will degrade the readability and overall experience. For example:
+
+![Scaling: No type scales ratio](https://miro.medium.com/max/4800/1*zH_frLw4eZCsBajEfcScHw.png)
+
+> [Read here](https://medium.com/sketch-app-sources/exploring-responsive-type-scales-cf1da541be54) for more info on the above two.
+
+### **Fluid Typography**
+This is viewport-based scaling and uses the `calc()` CSS function, which allows you to perform mathematical operations using a combination of different CSS units. Viewport units, specifically the viewport width (`vw`) units. For example:
+
+![Fluid Typography](https://blog.logrocket.com/wp-content/uploads/2019/10/minimum-font-size-nocdn.png)
+
+This type of scale is a bit complex and very dynamic, as per our use case we don't need that much of flexibility in typography across devices.
+
+> [Read here](https://blog.logrocket.com/the-elements-of-responsive-typography/) for more info on fluid typography.
+
+### **Custom Scaling**
+As the name suggests this type of scaling is preferred based on visual assumptions and is very custom in nature. The decisions around the type scale here doesn't hold any principles and are best suited for fancy marketing/landing pages.
+
+> [Read here](https://eightshapes.com/articles/typography-in-design-systems/) for more info on responsive typography.
+
+# Adoption strategy
+We need to implement this type-scale as a core part of DS. This will automatically drive adoption.
+- This won't impact the existing DS.
+- It is a part of the new DS and whenever the new DS is implemented the whole typography across all devices and pages will be affected.
```

Comment:
Add one more point
* This will be implemented as tokens in the new version of DS